FindTraining.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect } from "react";
+import { getAllTrainings } from "../../../Utils/DataManagement";
+import { useState } from "react";
+import useLocalStorage from "../../../Hooks/useLocalStorage";
+import { AVAILABLE_TRAINERS_HEADER } from "../../../Utils/StringLiterals";
+import {
+  eachDayOfInterval,
+  format,
+  startOfWeek,
+  endOfMonth,
+  endOfWeek,
+  add,
+  parse,
+  startOfTomorrow,
+} from "date-fns";
+import ScheduledTraining from "./ScheduledTrainers/ScheduledTraining";
+import { io } from "socket.io-client";
+import NavBar from "../../NavBar/NavBar";
+import "./FindTraining.css";
+import FindTrainingCalender from "./FindTrainingCalender";
+const socket = io("http://localhost:8900");
+
+export default function FindTraining() {
+  let tommorow = startOfTomorrow();
+  let [selectedDay, setSelectedDay] = useState(tommorow);
+  let [currentMonth, setCurrentMonth] = useState(format(tommorow, "MMM-yyyy"));
+  let firstDayCurrentMonth = parse(currentMonth, "MMM-yyyy", new Date());
+  let days = eachDayOfInterval({
+    start: startOfWeek(firstDayCurrentMonth),
+    end: endOfWeek(endOfMonth(firstDayCurrentMonth)),
+  });
+  function nextMonth() {
+    let firstDayCurrentMonth = parse(currentMonth, "MMM-yyyy", new Date());
+    let firstDayNextMonth = add(firstDayCurrentMonth, { months: 1 });
+    setCurrentMonth(format(firstDayNextMonth, "MMM-yyyy"));
+  }
+  function previousMonth() {
+    let firstDayCurrentMonth = parse(currentMonth, "MMM-yyyy", new Date());
+    let firstDayNextMonth = add(firstDayCurrentMonth, { months: -1 });
+    setCurrentMonth(format(firstDayNextMonth, "MMM-yyyy"));
+  }
+  const [allTrainings, setAllTrainings] = useLocalStorage("allTrainings", []);
+  useEffect(() => {
+    async function getTrainings() {
+      setAllTrainings(await getAllTrainings());
+    }
+    getTrainings();
+    socket.on("newTraining", (Training) => {
+      getTrainings();
+    });
+    socket.on("deletedTraining", (id) => {
+      const updatedTraining = allTrainings.filter((training) => {
```

Comment:
where are you using this updatedTraining?

Reply:
It is to return all the trainings that don't have the Id of the training that should be deleted. I have removed the function call getTrainings() from the socket.on and replaced it with setting the allTrainings to the filters training without the deleted training included